### PR TITLE
Predict Challenge README

### PR DESCRIPTION
**Pull request by freddyaboulton**
@emwolff @holger-nutonomy This is the first draft of the README explaining the rules of the predict challenge. Please review!

### CODE REVIEW

File: python-sdk/nuscenes/eval/predict/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- [Results format](#results-format)
+- [Evaluation metrics](#evaluation-metrics)
+
+## Introduction
+The goal of the nuScenes prediction task is to predict the future trajectories of objects in the nuScenes dataset.
+A trajectory is a sequence of x-y locations. For this challenge, the predictions are 6-seconds long and sampled at
+2 hertz.
+
+## Challenges
+To allow users to benchmark the performance of their method against the community, we will host a single leaderboard all-year round.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: I think it is "all year round."

File: python-sdk/nuscenes/eval/predict/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Additionally, we organize a number of challenges at leading Computer Vision and Machine Learning conference workshops.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: "we [will, intend to] organize"? Let's not over promise.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Users that submit their results during the challenge period are eligible for awards.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: Be clear that awards may be different for each challenge.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Any user that cannot attend the workshop (direct or via a representative) will be excluded from the challenge, but will still be listed on the leaderboard.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: The exact language might change for each challenge. I like the idea of requesting that people attend the workshop (possibly via representative) to win, though.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+### Workshop on Benchmarking Progress in Autonomous Driving, ICRA 2020
+The first nuScenes prediction challenge will be held at [ICRA 2020](https://www.icra2020.org/).
+The submission period will open in early march and continue until May 28th, 2020.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: March.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Results and winners will be announced at the [Workshop on Benchmarking Progress in Autonomous Driving](http://montrealrobotics.ca/driving-benchmarks/).
+Note that the evaluation server can still be used to benchmark your results after the challenge period.
+
+## Submission rules
+### Prediction-specific rules
+* The user can submit up to 25 proposed future trajectories, called `modes`, for each agent along with a probability the agent follows that proposal. Our metrics (explained below) will measure how well this proposed set of trajectories matches the ground truth.
+* Unlike previous challenges, the user will not submit their submissions to the eval server. Instead, we will run the user's model
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: Maybe add a note here on why we are doing this. The goal is to not expose the test set for other object detection challenges.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+on the test set for them. To make this possible, the user should modify the `do_inference_for_submission` function in the [inference script](https://github.com/nutonomy/nuscenes-devkit/blob/nuscenes-predict-challenge/python-sdk/nuscenes/eval/predict/do_inference.py)
+so that their model is used.
+* Up to two seconds of past history can be used to predict the future trajectory for each agent.
+* Every submission to the challenge must be accompanied by a technical report describing the method in sufficient detail to allow for independent verification.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: Want to expand on this a little bit. Right now, it is quite subjective and vague. We could have details on a separate page that we link to (e.g., do we need code, hyperparams for training, data augmentation techniques, model architecture diagrams, etc).

File: python-sdk/nuscenes/eval/predict/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+### General rules
+* We release annotations for the train and val set, but not for the test set.
+* We release sensor data for train, val and test set.
+* Top leaderboard entries and their papers will be manually reviewed.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: "Manually reviewed" for what exactly? That the tech report is sufficient to reproduce?

File: python-sdk/nuscenes/eval/predict/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+* Each user or team can have at most one one account on the evaluation server.
+* Each user or team can submit at most 3 results. These results must come from different models, rather than submitting results from the same model at different training epochs or with slightly different parameters.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: Hyperparams or model weights?

File: python-sdk/nuscenes/eval/predict/README.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+* Any attempt to circumvent these rules will result in a permanent ban of the team or company from all nuScenes challenges.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: Need to have clear, objective rules to enforce a "death penalty" like this.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+- [Participation](#participation)
+- [Challenges](#challenges)
+- [Submission rules](#submission-rules)
+- [Results format](#results-format)
+- [Evaluation metrics](#evaluation-metrics)
+
+## Introduction
+The goal of the nuScenes prediction task is to predict the future trajectories of objects in the nuScenes dataset.
+A trajectory is a sequence of x-y locations. For this challenge, the predictions are 6-seconds long and sampled at
+2 hertz.
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: I think we originally mentioned including everyone (cars, cyclists, pedestrians) but it would be better to do just cars since that's the data we tested our baselines on.

File: python-sdk/nuscenes/eval/predict/README.md (hunk 1 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Done

File: python-sdk/nuscenes/eval/predict/README.md (hunk 2 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Noted

File: python-sdk/nuscenes/eval/predict/README.md (hunk 3 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Added a sentence mentioning that!

File: python-sdk/nuscenes/eval/predict/README.md (hunk 6 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Done

File: python-sdk/nuscenes/eval/predict/README.md (hunk 7 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: I think we get their code automatically since we run their submission for them so I was thinking it'd be focused on explaining the rationale behind their code and their training procedure.

File: python-sdk/nuscenes/eval/predict/README.md (hunk 8 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Yes and that they didn't cheat somehow.

File: python-sdk/nuscenes/eval/predict/README.md (hunk 10 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: I think the idea here is that anyone caught trying to make more submissions than allowed will be banned. I will just say that instead.

File: python-sdk/nuscenes/eval/predict/README.md (hunk 9 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: I took these from the general rules for the prevous nuScenes challenges. I think the idea is to not have users overfit to the test set by allowing them infinite evals. I'll be explicit by explicitly mentioning "weights and hyperparameters" as opposed to just "parameters"

**Comment by freddyaboulton** [comment]
@emwolff Made some edits to reflect your comments.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 12]
```diff
@@ -15,33 +15,33 @@ A trajectory is a sequence of x-y locations. For this challenge, the predictions
...
 * We release sensor data for train, val and test set.
-* Top leaderboard entries and their papers will be manually reviewed.
+* Top leaderboard entries and their papers will be manually reviewed to ensure no cheating was done.
 * Each user or team can have at most one one account on the evaluation server.
-* Each user or team can submit at most 3 results. These results must come from different models, rather than submitting results from the same model at different training epochs or with slightly different parameters.
-* Any attempt to circumvent these rules will result in a permanent ban of the team or company from all nuScenes challenges.
+* Each user or team can submit at most 3 results. These results must come from different models, rather than submitting results from the same model at different training epochs or with slightly weights or hyperparameter values.
+* Any attempt to make more submissions than allowed will result in a permanent ban of the team or company from all nuScenes challenges.
 
 ## Results format
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/README.md]: Below we say `the prediction must be reported in the agent's local coordinate frame`. In nuScenes all annotations are in global coordinates. Wouldn't it be better to align that? It's also easier to plot on the map.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 13]
```diff
@@ -15,33 +15,33 @@ A trajectory is a sequence of x-y locations. For this challenge, the predictions
...
 The `do_inference_for_submission` function must produce a list of [`Predictions`](https://github.com/nutonomy/nuscenes-devkit/blob/nuscenes-predict-challenge/python-sdk/nuscenes/eval/predict/data_classes.py). A `Prediction` has the following components:
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/README.md]: Below you write "l2 distance". I suggest capitalizing it to "L2".

Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Thanks Holger!

File: python-sdk/nuscenes/eval/predict/README.md (hunk 12 above)
Comment (freddyaboulton) [reply python-sdk/nuscenes/eval/predict/README.md]: Sounds good. I will make this change.

**Comment by freddyaboulton** [comment]
@emwolff I changed the rules section to reflect what we discussed this morning. Let me know what you think.

File: python-sdk/nuscenes/eval/predict/README.md [hunk 14]
```diff
@@ -29,10 +29,8 @@ Note that the evaluation server can still be used to benchmark your results afte
...
-on the test set for them. This is because we do not want to interfere with the [nuScenes detection challenge](https://www.nuscenes.org/object-detection?externalData=all&mapData=all&modalities=Any) by exposing the test set annotations. In order for us to run their model, the user should modify the `do_inference_for_submission` function in the [inference script](https://github.com/nutonomy/nuscenes-devkit/blob/nuscenes-predict-challenge/python-sdk/nuscenes/eval/predict/do_inference.py)
-so that their model is used.
 * Up to two seconds of past history can be used to predict the future trajectory for each agent.
+* Unlike previous challenges, the leaderboard will be ranked according to performance on the nuScenes val set. This is because we cannot release the annotations on the test set, so users would not be able to run their models on the test set and then submit their predictions to the server. To prevent overfitting on the val set, the top 5 submissions on the leaderboard will be asked to send us their code and we will run their model on the test set. The winners will be chosen based on their performance on the test set, not the val set.
```
Comment (emwolff) [inline python-sdk/nuscenes/eval/predict/README.md]: "we cannot release the annotations on the test set" -> explain in parentheses that doing so would undercut nuScenes object prediction challenges we run.
